data_validation/combiner.py

```diff
@@ -101,9 +102,32 @@ def generate_report(
 
     result_df = client.execute(documented)
     result_df.validation_status.fillna(consts.VALIDATION_STATUS_FAIL, inplace=True)
+
+    first = next(iter(run_metadata.validations))
+    # get the first validation metadata object to fill source or target empty table names
+    result_df.source_table_name.fillna(
+        _get_table_name_when_null(
+            run_metadata.validations[first], consts.RESULT_TYPE_SOURCE
+        ),
+        inplace=True,
+    )
+    result_df.target_table_name.fillna(
+        _get_table_name_when_null(
+            run_metadata.validations[first], consts.RESULT_TYPE_TARGET
+        ),
+        inplace=True,
+    )
+
     return result_df
 
 
+def _get_table_name_when_null(
+    validation: metadata.ValidationMetadata, result_type: str
+) -> str:
+    value = validation.get_table_name(result_type)
+    return value if value is not None else np.nan
```

Comment:
@nehanene15 I just noticed later I could've used `float("nan")` here instead, let me know what do you think it's better

Reply:
We probably don't need another function for this since there isn't a case where this will be empty.
Instead we can just call `run_metadata.validations[first].get_table_name(consts.RESULT_TYPE_SOURCE)`

Reply:
That was the initial approach I implemented that day and I thought the same, but actually the table names can be empty for custom-query validations, that's why a Spanner system test was failing